### ISSUE TITLE
Piping currently requires that the `-threaded` flag be passed to the compiler (undocumented).

### ISSUE THREAD

**Issue by maralorn** [luke-clifton/shh#36]
I am writing some mail-filtering tools with the mblaze suit and shh. Those tools often consume stdin and output something on stdout.
Sadly I encounter regularly a weird kind of bug where various external commands called by shh do not terminate. It is somehow highly dependend on context and horrible to debug. When reproducible in shell the shell version always works. Even trying to build a minimal example in ghci I can never reproduce the bug. Nevertheless I had the bug multiple times in my tool and when a version of the code does not work it does not work consistently given the same input.
Now I have never encountered the bug without some kind of piping being involved. So I guess it has something to do with pipes (presumably stdin?) not being closed correctly. But that doesn‘t seem to explain it. Maybe it has something to do with stdout buffering? I am a bit lost …
But since I can‘t reproduce this in shell it might be specific to something shh does or does not do. Do you have any insights into how such a problem could occur and how I could work around this?

Edit:
I don‘t think this helps much put just giving an example of code that hangs in the call to e.g. `mhdr` a tool that prints mailheaders.
```haskell
tos <- mhdr "-h" "To" "-d" "Maildir" "-A" |> capture                                                                                                                                                                                                                                                   
print tos
```

**Comment by maralorn** [comment]
Oookay. This cost me the better part of my afternoon.

Solution: You need to compile with `-threaded`.

Do you think that‘s right? If yes do you think it would be a good idea to add a warning for this to the readme?

**Comment by luke-clifton** [comment]
Yes, unfortunately we currently need `-threaded` in many scenarios.

The issue is that `wait` blocks. I am very keen to remove that restriction, and it is possible to do it if I'm willing to modify the state of the process (install signal handlers).

I'm thinking about what the best way to handle this is, but currently, the answer is to use `-threaded`.

I'll leave this issue open to remind me to make it more clear in the documentation.

**Comment by luke-clifton** [comment]
Fixed in c03d9e5924b0d9b5c325ecd8dc3ead901c007f4e
